### DISCUSSION TITLE
issue with building the test after update

### DISCUSSION THREAD

**Discussion by Khaksarkn** [idaholab/moose#24342] [Q&A Getting Started] [answered]
Hi, I have an m1 macbook air and had a working moose. But recently when I created new application, running the first test gave me segmentation error. So, I updated moose and now I cannot even make the tests in the moose directory to test if moose works right. this is the error I get when I try to make in the /projects/moose/test/ directory:
I tried installing packaging (pip install packaging and python -m ensurepip --upgrade) but it already has the packaging.

I would appreciate it if you could help me out. Thanks.

I also tried the steps in https://mooseframework.inl.gov/help/troubleshooting.html#buildissues
and got the mentioned outputs. but this error still is:

Traceback (most recent call last):
  File "/Users/kiankhaksar/projects/moose/framework/scripts/get_repo_revision.py", line 19, in <module>
    from packaging import version
ModuleNotFoundError: No module named 'packaging'
Checking if header needs updating: /Users/kiankhaksar/projects/moose/framework/include/base/MooseRevision.h...

Failed to generate MooseRevision.h

make: *** [/Users/kiankhaksar/projects/moose/framework/moose.mk:347: /Users/kiankhaksar/projects/moose/framework/include/base/MooseRevision.h] Error 1

**Comment by GiudGiud** [comment]
Hello

Are you not using mamba?
Why are you using pip?

Guillaume

**Reply by Khaksarkn** [reply]
Hi, 
Thanks for the reply.

Yes, I am using mamba. I searched for possible solutions. they suggested insalling packaging this way. How may I resolve the issue?

Thanks.

**Reply by Khaksarkn** [reply]
I just tried (mamba install packaging). But I am still getting the same thing.

**Reply by GiudGiud** [reply]
I'm worried there's different python being found. 
What does `which python` return ?

**Reply by Khaksarkn** [reply]
This is the output:
/Users/kiankhaksar/mambaforge3/envs/moose/bin/python

**Reply by GiudGiud** [reply]
in fact what does the diagnostics script in moose/scripts return?

**Reply by GiudGiud** [reply]
> This is the output: /Users/kiankhaksar/mambaforge3/envs/moose/bin/python

looks good

**Reply by Khaksarkn** [reply]
I run the diagnostics.sh file in the terminal and this is the output:

Fri 12 May 2023 14:27:32 EDT

System Arch: Darwin Kernel Version 22.4.0: Mon Mar 6 21:00:41 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T8103

MOOSE Package Version: Custom Build

CPU Count: 8

Memory Free: 335.703 MB

Variable `which $CC` check:
/Users/kiankhaksar/mambaforge3/envs/moose/bin/mpicc

$CC --version:
clang version 12.0.1
Target: arm64-apple-darwin20.0.0
Thread model: posix
InstalledDir: /Users/kiankhaksar/mambaforge3/envs/moose/bin

MPICC:
which mpicc:
	/Users/kiankhaksar/mambaforge3/envs/moose/bin/mpicc
mpicc -show:
	arm64-apple-darwin20.0.0-clang -I/Users/kiankhaksar/mambaforge3/envs/moose/include -I/Users/kiankhaksar/mambaforge3/envs/moose/include -L/Users/kiankhaksar/mambaforge3/envs/moose/lib -Wl,-rpath,/Users/kiankhaksar/mambaforge3/envs/moose/lib -Wl,-commons,use_dylibs -I/Users/kiankhaksar/mambaforge3/envs/moose/include -L/Users/kiankhaksar/mambaforge3/envs/moose/lib -lmpi -lpmpi

COMPILER arm64-apple-darwin20.0.0-clang:
clang version 12.0.1
Target: arm64-apple-darwin20.0.0
Thread model: posix
InstalledDir: /Users/kiankhaksar/mambaforge3/envs/moose/bin

Python:
	/Users/kiankhaksar/mambaforge3/envs/moose/bin/python
	Python 3.10.8

MODULES NOT AVAILABLE

PETSc configure:
static const char *petscconfigureoptions = "--download-hypre=1 --with-shared-libraries=1 --with-hdf5-dir=${PREFIX}  --with-make-np=8  --with-debugging=no --download-fblaslapack=1 --download-metis=1 --download-ptscotch=1 --download-parmetis=1 --download-superlu_dist=1 --download-mumps=1 --download-strumpack=1 --download-scalapack=1 --download-slepc=1 --with-mpi=1 --with-openmp=1 --with-cxx-dialect=C++11 --with-fortran-bindings=0 --with-sowing=0 --with-64-bit-indices --COPTFLAGS=-O3 --CXXOPTFLAGS=-O3 --FOPTFLAGS=-O3 --with-x=0 --with-ssl=0 CC=mpicc CXX=mpicxx FC=mpif90 FC=mpif90 FC=mpif77 AR=arm64-apple-darwin20.0.0-ar RANLIB=arm64-apple-darwin20.0.0-ranlib CFLAGS="-ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem /Users/kiankhaksar/mambaforge3/envs/moose/include   -mcpu=apple-a12" CXXFLAGS="-ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -fmessage-length=0 -isystem ${PREFIX}/include -std=c++17 -mcpu=apple-a12" CPPFLAGS="-D_FORTIFY_SOURCE=2 -isystem /Users/kiankhaksar/mambaforge3/envs/moose/include -mmacosx-version-min=11.3" FFLAGS="-march=armv8.3-a -ftree-vectorize -fPIC -fno-stack-protector -O2 -pipe -isystem /Users/kiankhaksar/mambaforge3/envs/moose/include   -march=armv8.3-a" FCFLAGS="-march=armv8.3-a -ftree-vectorize -fPIC -fno-stack-protector -O2 -pipe -isystem /Users/kiankhaksar/mambaforge3/envs/moose/include   -march=armv8.3-a" LDFLAGS="-Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath,/Users/kiankhaksar/mambaforge3/envs/moose/lib -L/Users/kiankhaksar/mambaforge3/envs/moose/lib" --prefix=/Users/kiankhaksar/mambaforge3/envs/moose";

PETSc linked libraries:
/Users/kiankhaksar/mambaforge3/envs/moose/lib/libpetsc.dylib:
	@rpath/libpetsc.3.16.dylib (compatibility version 3.16.0, current version 3.16.6)
	@rpath/libHYPRE-2.23.0.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libstrumpack.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libsuperlu_dist.7.dylib (compatibility version 7.0.0, current version 7.1.1)
	@rpath/libhdf5.200.dylib (compatibility version 202.0.0, current version 202.0.0)
	@rpath/libparmetis.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libmetis.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libc++.1.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.100.5)
	@rpath/libmpifort.12.dylib (compatibility version 15.0.0, current version 15.2.0)
	@rpath/libmpi.12.dylib (compatibility version 15.0.0, current version 15.2.0)
	@rpath/libgfortran.5.dylib (compatibility version 6.0.0, current version 6.0.0)
	@rpath/libomp.dylib (compatibility version 5.0.0, current version 5.0.0)

ENVIRONMENT:
AS=arm64-apple-darwin20.0.0-as
MANPATH=/opt/homebrew/share/man:
AR=arm64-apple-darwin20.0.0-ar
LDFLAGS=-Wl,-pie -Wl,-headerpad_max_install_names -Wl,-dead_strip_dylibs -Wl,-rpath,/Users/kiankhaksar/mambaforge3/envs/moose/lib -L/Users/kiankhaksar/mambaforge3/envs/moose/lib
ac_cv_func_malloc_0_nonnull=yes
MOOSE_NO_CODESIGN=true
TERM_PROGRAM=Apple_Terminal
VTKINCLUDE_DIR=/Users/kiankhaksar/mambaforge3/envs/moose/libmesh-vtk/include/vtk-9.2
HDF5_DIR=/Users/kiankhaksar/mambaforge3/envs/moose
ac_cv_func_realloc_0_nonnull=yes
TERM=xterm-256color
SHELL=/bin/bash
HOST=arm64-apple-darwin20.0.0
OBJC=arm64-apple-darwin20.0.0-clang
NM=arm64-apple-darwin20.0.0-nm
TMPDIR=/var/folders/z1/fk6267bx7bqdpn5b8g2ry23w0000gn/T/
HOMEBREW_REPOSITORY=/opt/homebrew
CPPFLAGS=-D_FORTIFY_SOURCE=2 -isystem /Users/kiankhaksar/mambaforge3/envs/moose/include
CONDA_SHLVL=2
PAGESTUFF=arm64-apple-darwin20.0.0-pagestuff
TERM_PROGRAM_VERSION=447
CONDA_PROMPT_MODIFIER=(moose) 
LIPO=arm64-apple-darwin20.0.0-lipo
NMEDIT=arm64-apple-darwin20.0.0-nmedit
SIZE=arm64-apple-darwin20.0.0-size
GSETTINGS_SCHEMA_DIR_CONDA_BACKUP=
SDKROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk
TERM_SESSION_ID=08866D55-8C7E-443D-BC98-59D868116CA9
INSTALL_NAME_TOOL=arm64-apple-darwin20.0.0-install_name_tool
CXX_FOR_BUILD=/Users/kiankhaksar/mambaforge3/envs/moose/bin/arm64-apple-darwin20.0.0-clang++
GFORTRAN=/Users/kiankhaksar/mambaforge3/envs/moose/bin/arm64-apple-darwin20.0.0-gfortran
SEGEDIT=arm64-apple-darwin20.0.0-segedit
USER=kiankhaksar
SEG_HACK=arm64-apple-darwin20.0.0-seg_hack
LIBTOOL=arm64-apple-darwin20.0.0-libtool
CLANGXX=arm64-apple-darwin20.0.0-clang++
Eigen3_DIR=/Users/kiankhaksar/mambaforge3/envs/moose/libmesh/include/Eigen
CONDA_EXE=/Users/kiankhaksar/mambaforge3/bin/conda
FC_FOR_BUILD=/Users/kiankhaksar/mambaforge3/envs/moose/bin/arm64-apple-darwin20.0.0-gfortran
CONDA_TOOLCHAIN_BUILD=arm64-apple-darwin20.0.0
OTOOL=arm64-apple-darwin20.0.0-otool
WASP_DIR=/Users/kiankhaksar/mambaforge3/envs/moose/wasp
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.P5UEBmFNyE/Listeners
DEBUG_FORTRANFLAGS=-march=armv8.3-a -ftree-vectorize -fPIC -fno-stack-protector -O2 -pipe -isystem /Users/kiankhaksar/mambaforge3/envs/moose/include -march=armv8.3-a -ftree-vectorize -fPIC -fno-stack-protector -O2 -pipe -Og -g -Wall -Wextra -fcheck=all -fbacktrace -fimplicit-none -fvar-tracking-assignments
CXXFLAGS=-ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -fmessage-length=0 -isystem /Users/kiankhaksar/mambaforge3/envs/moose/include -std=c++17
CONDA_TOOLCHAIN_HOST=arm64-apple-darwin20.0.0
STRINGS=arm64-apple-darwin20.0.0-strings
_CE_CONDA=
CONDA_PREFIX_1=/Users/kiankhaksar/mambaforge3
CHECKSYMS=arm64-apple-darwin20.0.0-checksyms
PATH=/opt/homebrew/bin:/opt/homebrew/sbin:/opt/miniconda/bin:/Users/kiankhaksar/mambaforge3/envs/moose/bin:/Users/kiankhaksar/mambaforge3/condabin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Users/kiankhaksar/mambaforge3/envs/moose/wasp/bin:/Users/kiankhaksar/mambaforge3/envs/moose/wasp/bin:/Users/kiankhaksar/mambaforge3/envs/moose/wasp/bin:/Users/kiankhaksar/mambaforge3/envs/moose/wasp/bin:/Users/kiankhaksar/mambaforge3/envs/moose/wasp/bin
MESON_ARGS=--buildtype release
GSETTINGS_SCHEMA_DIR=/Users/kiankhaksar/mambaforge3/envs/moose/share/glib-2.0/schemas
XML_CATALOG_FILES=file:///Users/kiankhaksar/mambaforge3/envs/moose/etc/xml/catalog file:///etc/xml/catalog
__CFBundleIdentifier=com.apple.Terminal
CONDA_PREFIX=/Users/kiankhaksar/mambaforge3/envs/moose
LD=arm64-apple-darwin20.0.0-ld
BUILD=arm64-apple-darwin20.0.0
DEBUG_CXXFLAGS=-ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -fmessage-length=0 -Og -g -Wall -Wextra -isystem /Users/kiankhaksar/mambaforge3/envs/moose/include
C_INCLUDE_PATH=/Users/kiankhaksar/mambaforge3/envs/moose/include
PWD=/Users/kiankhaksar/projects/moose
F90=mpif90
STRIP=arm64-apple-darwin20.0.0-strip
LIBMESH_DIR=/Users/kiankhaksar/mambaforge3/envs/moose/libmesh
CMAKE_ARGS=-DCMAKE_AR=/Users/kiankhaksar/mambaforge3/envs/moose/bin/arm64-apple-darwin20.0.0-ar -DCMAKE_CXX_COMPILER_AR=/Users/kiankhaksar/mambaforge3/envs/moose/bin/arm64-apple-darwin20.0.0-ar -DCMAKE_C_COMPILER_AR=/Users/kiankhaksar/mambaforge3/envs/moose/bin/arm64-apple-darwin20.0.0-ar -DCMAKE_RANLIB=/Users/kiankhaksar/mambaforge3/envs/moose/bin/arm64-apple-darwin20.0.0-ranlib -DCMAKE_CXX_COMPILER_RANLIB=/Users/kiankhaksar/mambaforge3/envs/moose/bin/arm64-apple-darwin20.0.0-ranlib -DCMAKE_C_COMPILER_RANLIB=/Users/kiankhaksar/mambaforge3/envs/moose/bin/arm64-apple-darwin20.0.0-ranlib -DCMAKE_LINKER=/Users/kiankhaksar/mambaforge3/envs/moose/bin/arm64-apple-darwin20.0.0-ld -DCMAKE_STRIP=/Users/kiankhaksar/mambaforge3/envs/moose/bin/arm64-apple-darwin20.0.0-strip -DCMAKE_INSTALL_NAME_TOOL=/Users/kiankhaksar/mambaforge3/envs/moose/bin/arm64-apple-darwin20.0.0-install_name_tool -DCMAKE_LIBTOOL=/Users/kiankhaksar/mambaforge3/envs/moose/bin/arm64-apple-darwin20.0.0-libtool -DCMAKE_OSX_DEPLOYMENT_TARGET= -DCMAKE_BUILD_TYPE=Release -DCMAKE_OSX_SYSROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk
LANG=en_CA.UTF-8
F95=/Users/kiankhaksar/mambaforge3/envs/moose/bin/arm64-apple-darwin20.0.0-gfortran
XPC_FLAGS=0x0
DEBUG_FFLAGS=-march=armv8.3-a -ftree-vectorize -fPIC -fno-stack-protector -O2 -pipe -isystem /Users/kiankhaksar/mambaforge3/envs/moose/include -march=armv8.3-a -ftree-vectorize -fPIC -fno-stack-protector -O2 -pipe -Og -g -Wall -Wextra -fcheck=all -fbacktrace -fimplicit-none -fvar-tracking-assignments
F77=mpif77
REDO_PREBINDING=arm64-apple-darwin20.0.0-redo_prebinding
XPC_SERVICE_NAME=0
_CE_M=
CLANG=arm64-apple-darwin20.0.0-clang
CC_FOR_BUILD=/Users/kiankhaksar/mambaforge3/envs/moose/bin/arm64-apple-darwin20.0.0-clang
CXX=mpicxx
HOME=/Users/kiankhaksar
SHLVL=2
FORTRANFLAGS=-march=armv8.3-a -ftree-vectorize -fPIC -fno-stack-protector -O2 -pipe -isystem /Users/kiankhaksar/mambaforge3/envs/moose/include
HOMEBREW_PREFIX=/opt/homebrew
CFLAGS=-ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem /Users/kiankhaksar/mambaforge3/envs/moose/include
FC=mpif90
_CONDA_PYTHON_SYSCONFIGDATA_NAME=_sysconfigdata_arm64_apple_darwin20_0_0
MPIHOME=/Users/kiankhaksar/mambaforge3/envs/moose
VTKLIB_DIR=/Users/kiankhaksar/mambaforge3/envs/moose/libmesh-vtk/lib
LOGNAME=kiankhaksar
CONDA_PYTHON_EXE=/Users/kiankhaksar/mambaforge3/bin/python
build_alias=arm64-apple-darwin20.0.0
CONDA_DEFAULT_ENV=moose
RANLIB=arm64-apple-darwin20.0.0-ranlib
DEBUG_CFLAGS=-ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -Og -g -Wall -Wextra -isystem /Users/kiankhaksar/mambaforge3/envs/moose/include
HOMEBREW_CELLAR=/opt/homebrew/Cellar
INFOPATH=/opt/homebrew/share/info:
SEG_ADDR_TABLE=arm64-apple-darwin20.0.0-seg_addr_table
CC=mpicc
CMAKE_PREFIX_PATH=:/Users/kiankhaksar/mambaforge3/envs/moose
host_alias=arm64-apple-darwin20.0.0
LDFLAGS_LD=-pie -headerpad_max_install_names -dead_strip_dylibs -rpath /Users/kiankhaksar/mambaforge3/envs/moose/lib -L/Users/kiankhaksar/mambaforge3/envs/moose/lib
PETSC_DIR=/Users/kiankhaksar/mambaforge3/envs/moose
FFLAGS=-march=armv8.3-a -ftree-vectorize -fPIC -fno-stack-protector -O2 -pipe -isystem /Users/kiankhaksar/mambaforge3/envs/moose/include
OBJC_FOR_BUILD=/Users/kiankhaksar/mambaforge3/envs/moose/bin/arm64-apple-darwin20.0.0-clang
_=/usr/bin/env

**Reply by GiudGiud** [reply, marked as answer]
`PATH=/opt/homebrew/bin` this is likely the problem

Let s shuffle this up, get conda before homebrew
export PATH="/Users/kiankhaksar/mambaforge3/envs/moose/bin:/Users/kiankhaksar/mambaforge3/condabin:$PATH"

**Reply by Khaksarkn** [reply]
It started to build! and the tests are being OK'ed! still running!

Thank you very much. I really appreciate it.

So, using this command, I added the the paths to moose/bin and mambaforge3/condabin before the path to homebrew? since the program was accessing the python of homebrew before the python of moose? I'm guessing!

Is there a way of permanently changing the PATH ? 

Thank you for your help.

**Reply by GiudGiud** [reply]
Yes homebrew was (is) briniging in another python

You can add this command to your .bash_profile

**Reply by Khaksarkn** [reply]
Thank you very much. I really appreciate the help.
